sdk/storage/AzureStorageBlob/Source/TransferManager.swift

```diff
@@ -50,7 +50,8 @@ internal protocol TransferManager {
     func cancel(transfer: TransferImpl)
     func pause(transfer: TransferImpl)
     func remove(transfer: TransferImpl)
-    func resume(transfer: TransferImpl)
+    // TODO: Resolve issue where handler can't be downcast
+    func resume(transfer: TransferImpl, progressHandler: ((BlobTransfer) -> Void)?)
```

Comment:
Could not cast a `((BlobTransfer) -> Void)` closure to a `((TransferImpl) -> Void)` closure, so this is what I had to do just to get it to work for the time being.

Reply:
I initially considered a conversion function that would wrap a ` ((BlobTransfer) -> Void)` in another closure of type `((TransferImpl) -> Void)` and just do the cast internally, but I don't think that's the right path either. This is fine for now. I think the better approach down the road might involve generics of type `T: TransferImpl` rather than just taking a `TransferImpl`, especially since we switch on the *actual* type of the `TransferImpl` in a lot of our method bodies.